[BUG] global.d.ts doesn't compile

### Environment

🪟 Windows

### System

Windows

### Version

1.13.3

### Desktop Information

- Branch: release

### Describe the problem

[public/global.d.ts](https://github.com/SillyTavern/SillyTavern/blob/12ac17197925ee1e1dba00a9505001e09e13dfde/public/global.d.ts) just doesn't compile. 

Here's the result of `tsc --target esnext --module esnext --moduleResolution node --allowUmdGlobalAccess --allowSyntheticDefaultImports --strictBindCallApply public/global.d.ts` (flags matching the content of public/jsconfig.json) on the repo (using an external typescript@5.9.2) after `npm install` to get @types packages:
```
public/global.d.ts:8:5 - error TS1038: A 'declare' modifier cannot be used in an already ambient context.

8     declare type InstructSettings = typeof power_user.instruct;
      ~~~~~~~

public/global.d.ts:9:5 - error TS1038: A 'declare' modifier cannot be used in an already ambient context.

9     declare type ContextSettings = typeof power_user.context;
      ~~~~~~~

public/global.d.ts:10:5 - error TS1038: A 'declare' modifier cannot be used in an already ambient context.

10     declare type ReasoningSettings = typeof power_user.reasoning;
       ~~~~~~~

public/global.d.ts:17:5 - error TS1038: A 'declare' modifier cannot be used in an already ambient context.

17     declare var pdfjsLib;
       ~~~~~~~

public/global.d.ts:18:5 - error TS1038: A 'declare' modifier cannot be used in an already ambient context.

18     declare var ePub;
       ~~~~~~~

public/global.d.ts:19:5 - error TS1038: A 'declare' modifier cannot be used in an already ambient context.

19     declare var quickReplyApi: QuickReplyApi;
       ~~~~~~~

public/global.d.ts:21:5 - error TS1038: A 'declare' modifier cannot be used in an already ambient context.

21     declare var SillyTavern: {
       ~~~~~~~

public/global.d.ts:66:5 - error TS1040: 'async' modifier cannot be used in an ambient context.

66     async function translate(text: string, lang: string, provider: string = null): Promise<string>;
       ~~~~~

public/global.d.ts:66:58 - error TS2371: A parameter initializer is only allowed in a function or constructor implementation.

66     async function translate(text: string, lang: string, provider: string = null): Promise<string>;
                                                            ~~~~~~~~~~~~~~~~~~~~~~~

public/global.d.ts:79:15 - error TS2430: Interface 'ColorPickerEvent' incorrectly extends interface 'ChangeEvent<HTMLElement, any, any, any>'.
  Types of property 'detail' are incompatible.
    Type '{ rgba: string; }' is not assignable to type 'undefined'.

79     interface ColorPickerEvent extends JQuery.ChangeEvent<HTMLElement> {
                 ~~~~~~~~~~~~~~~~


Found 10 errors in the same file, starting at: public/global.d.ts:8
```

The [instructions for writing UI extensions](https://docs.sillytavern.app/for-contributors/writing-extensions/#typescript-notice) say that you should `import` public/global.d.ts to get autocomplete and so forth, but incorporating public/global.d.ts into the compilation causes the compile to fail, as shown above.

(I am writing a UI extension for my personal use in TypeScript, and have had to locally patch these compilation errors out of public/global.d.ts to get things working. I have to imagine that nobody else is trying to write TypeScript extension code that references public/global.d.ts, or else someone would have raised this issue by now.)

### Additional info

_No response_

### Please tick the boxes

- [x] I have explained the issue clearly, and I included all relevant info
- [x] I have checked that this [issue hasn't already been raised](https://github.com/SillyTavern/SillyTavern/issues?q=is%3Aissue)
- [x] I have checked the [docs](https://docs.sillytavern.app/) ![important](https://img.shields.io/badge/Important!-F6094E)
- [x] I confirm that my issue is not related to third-party content, unofficial extension or patch. If in doubt, check with a new [user account](https://docs.sillytavern.app/administration/multi-user/) and with extensions disabled

--- a/public/global.d.ts
+++ b/public/global.d.ts
@@ -5,20 +5,20 @@ import { QuickReplyApi } from './scripts/extensions/quick-reply/api/QuickReplyAp
 
 declare global {
     // Custom types
-    export type InstructSettings = typeof power_user.instruct;
-    export type ContextSettings = typeof power_user.context;
-    export type ReasoningSettings = typeof power_user.reasoning;
+    type InstructSettings = typeof power_user.instruct;
+    type ContextSettings = typeof power_user.context;
+    type ReasoningSettings = typeof power_user.reasoning;
 
     // Global namespace modules
     interface Window {
         ai: any;
     }
 
-    export var pdfjsLib;
-    export var ePub;
-    export var quickReplyApi: QuickReplyApi;
+    var pdfjsLib;
+    var ePub;
+    var quickReplyApi: QuickReplyApi;
 
-    export var SillyTavern: {
+    var SillyTavern: {
         getContext(): typeof getContext;
         llm: any;
         libs: typeof libs;

--- a/public/global.d.ts
+++ b/public/global.d.ts
@@ -63,7 +63,7 @@ declare global {
      * @param lang Target language
      * @param provider Translation provider
      */
-    function translate(text: string, lang: string, provider: string = null): Promise<string>;
+    function translate(text: string, lang: string, provider?: string | null): Promise<string>;
 
     interface ConvertVideoArgs {
         buffer: Uint8Array;

--- a/public/global.d.ts
+++ b/public/global.d.ts
@@ -63,7 +63,7 @@ declare global {
      * @param lang Target language
      * @param provider Translation provider
      */
-    async function translate(text: string, lang: string, provider: string = null): Promise<string>;
+    function translate(text: string, lang: string, provider: string = null): Promise<string>;
 
     interface ConvertVideoArgs {
         buffer: Uint8Array;

--- a/public/global.d.ts
+++ b/public/global.d.ts
@@ -76,9 +76,9 @@ declare global {
      */
     function convertVideoToAnimatedWebp(args: ConvertVideoArgs): Promise<Uint8Array>;
 
-    interface ColorPickerEvent extends JQuery.ChangeEvent<HTMLElement> {
+    type ColorPickerEvent = Omit<JQuery.ChangeEvent<HTMLElement>, "detail"> & {
         detail: {
             rgba: string;
-        };
-    }
+        }
+    };
 }

--- a/public/global.d.ts
+++ b/public/global.d.ts
@@ -5,20 +5,20 @@ import { QuickReplyApi } from './scripts/extensions/quick-reply/api/QuickReplyAp
 
 declare global {
     // Custom types
-    declare type InstructSettings = typeof power_user.instruct;
-    declare type ContextSettings = typeof power_user.context;
-    declare type ReasoningSettings = typeof power_user.reasoning;
+    export type InstructSettings = typeof power_user.instruct;
+    export type ContextSettings = typeof power_user.context;
+    export type ReasoningSettings = typeof power_user.reasoning;
 
     // Global namespace modules
     interface Window {
         ai: any;
     }
 
-    declare var pdfjsLib;
-    declare var ePub;
-    declare var quickReplyApi: QuickReplyApi;
+    export var pdfjsLib;
+    export var ePub;
+    export var quickReplyApi: QuickReplyApi;
 
-    declare var SillyTavern: {
+    export var SillyTavern: {
         getContext(): typeof getContext;
         llm: any;
         libs: typeof libs;

